ZuluContent/Spells/Sixth/Reveal.cs

```diff
@@ -1,90 +1,57 @@
-using System.Collections.Generic;
-using Server.Targeting;
+using System.Threading.Tasks;
+using Server.Network;
 
 namespace Server.Spells.Sixth
 {
-    public class RevealSpell : MagerySpell
+    public class RevealSpell : MagerySpell, IAsyncSpell
     {
-        public RevealSpell(Mobile caster, Item spellItem) : base(caster, spellItem)
-        {
-        }
-
+        public RevealSpell(Mobile caster, Item spellItem) : base(caster, spellItem) { }
 
-        public override void OnCast()
+        public async Task CastAsync()
         {
-            Caster.Target = new InternalTarget(this);
-        }
-
-        public void Target(IPoint3D p)
-        {
-            if (!Caster.CanSee(p))
-            {
-                Caster.SendLocalizedMessage(500237); // Target can not be seen.
-            }
-            else if (CheckSequence())
-            {
-                SpellHelper.Turn(Caster, p);
+            var target = SpellHelper.GetSurfaceTop(Caster.Location);
 
-                SpellHelper.GetSurfaceTop(ref p);
+            var range = (int) (Caster.Skills[SkillName.Magery].Value / 10.0 - 5);
 
-                var targets = new List<Mobile>();
-
-                var map = Caster.Map;
-
-                if (map != null)
-                {
-                    IPooledEnumerable eable = map.GetMobilesInRange(new Point3D(p),
-                        1 + (int) (Caster.Skills[SkillName.Magery].Value / 20.0));
-
-                    foreach (Mobile m in eable)
-                        if (m.Hidden && (m.AccessLevel == AccessLevel.Player || Caster.AccessLevel > m.AccessLevel) &&
-                            CheckDifficulty(Caster, m))
-                            targets.Add(m);
-
-                    eable.Free();
-                }
+            var found = 0;
+            var eable = Caster.Map.GetMobilesInRange(target, range > 1 ? range : 1);
+            foreach (var hider in eable)
+            {
+                if (!hider.Hidden)
+                    continue;
+                
+                var skill = SpellHelper.TryResistDamage(Caster, hider, Circle, (int)Caster.Skills[SkillName.Magery].Value);
+                var resist = hider.Skills[SkillName.MagicResist].Value / 1.5;
```

Comment:
wow always thought Hiding skill would also play a role in magic reveal

Reply:
Yeah it's kinda broken IMO, thieves get revealed by any level of magery.